src/controllers/image/index.ts

```diff
@@ -1,42 +1,60 @@
 import type { FastifyInstance } from "fastify";
 
-import { BadRequestError } from "@errors";
-import { OpenAIClient } from "@helpers";
+import { BadRequestError, InternalServerError } from "@errors";
+import { OpenAIClient, slugify } from "@helpers";
+import { openaiBucket } from "@aws/buckets/openai";
+import { users, openAiImages } from "@src/models";
 
 export const imageRouter = async function (app: FastifyInstance) {
-	app.post<{ Body: { prompt: string } }>(
+	app.post<{ Body: { prompt: string; user_id: string; server_id: string } }>(
 		"/image",
 		{
 			preValidation: [app.restricted],
 			schema: {
 				body: {
 					type: "object",
-					required: ["prompt"],
+					required: ["prompt", "user_id", "server_id"],
 					additionalProperties: false,
 					properties: {
-						prompt: { type: "string" }
+						prompt: { type: "string" },
+						user_id: { type: "string" },
+						server_id: { type: "string" }
 					}
 				},
 				response: {
-					200: {
-						type: "object",
-						properties: {
-							image_url: { type: "string" }
-						}
-					}
+					200: { $ref: "openaiChallenge#" }
 				}
 			}
 		},
 		async (req) => {
 			try {
-				const { prompt } = req.body;
+				const { prompt, server_id, user_id } = req.body;
+				await users.assertExists({ user_id, server_id });
+				const filename = slugify(`${user_id} ${server_id} ${prompt}.png`);
+				if (filename.length >= 1024) {
+					throw new BadRequestError("input prompt too long");
+				}
 				const response = await OpenAIClient.createImage({
 					prompt,
 					n: 1,
-					size: "512x512"
+					size: "512x512",
```

Comment:
1024x1024 is the largest option, I generated one and it is about 3 MB. Do we have enough room on S3 to bump it up?